src/mlir/dialect/VeronaOps.cc

```diff
@@ -113,6 +114,25 @@ static LogicalResult verify(verona::ClassOp classOp)
 
 namespace mlir::verona
 {
+  LogicalResult CopyOp::typecheck()
+  {
+    return checkSubtype(getLoc(), input().getType(), output().getType());
+  }
+
+  LogicalResult ReturnOp::typecheck()
+  {
+    auto fn = cast<FuncOp>(getParentOp());
+    const auto& results = fn.getType().getResults();
+    if (results.size() != 1)
+    {
+      // TODO: The `verona.return` op could easily be extended to support
```

Comment:
`return` can actually return 0 and multiple values.

The generator has a class `ReturnValue` that handles that, too.

Reply:
Done.